src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,977 @@
+                                        // Playlist to track mapping found, update map
+                                        rekordbox_pdb_t::playlist_entry_row_t *playlistEntry =
+                                                (rekordbox_pdb_t::playlist_entry_row_t *)(*rowRef)->body();
+                                        playlistTrackMap[playlistEntry->playlist_id()][playlistEntry->entry_index()] =
+                                                playlistEntry->track_id();
+                                    } break;
+                                    case rekordbox_pdb_t::PAGE_TYPE_TRACKS: {
+                                        // Track found, insert into database
+                                        insertTrack(
+                                                database, (rekordbox_pdb_t::track_row_t *)(*rowRef)->body(), query, queryInsertIntoDevicePlaylistTracks, artistsMap, albumsMap, genresMap, keysMap, devicePath, device, audioFilesCount);
+
+                                        audioFilesCount++;
+                                    } break;
+                                    case rekordbox_pdb_t::PAGE_TYPE_PLAYLIST_TREE: {
+                                        // Playlist tree node found, update map
+                                        rekordbox_pdb_t::playlist_tree_row_t *playlistTree =
+                                                (rekordbox_pdb_t::playlist_tree_row_t *)(*rowRef)->body();
+
+                                        playlistNameMap[playlistTree->id()] = getText(playlistTree->name());
+                                        playlistIsFolderMap[playlistTree->id()] = playlistTree->is_folder();
+                                        playlistTreeMap[playlistTree->parent_id()][playlistTree->sort_order()] = playlistTree->id();
+
+                                        folderOrPlaylistFound = true;
+                                    } break;
+                                    default:
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    if (currentRef->index() == lastIndex) {
+                        moreLeft = false;
+                    } else {
+                        currentRef = page->next_page();
+                    }
+                } while (moreLeft);
+                done = true;
+            }
+        }
+    }
+
+    if (audioFilesCount > 0 || folderOrPlaylistFound) {
+        // If we have found anything, recursively build playlist/folder TreeItem children
+        // for the original device TreeItem
+        buildPlaylistTree(database, deviceItem, 0, playlistNameMap, playlistIsFolderMap, playlistTreeMap, playlistTrackMap, devicePath, device);
+    }
+
+    qDebug() << "Found: " << audioFilesCount << " audio files in Rekordbox device " << device;
+
+    transaction.commit();
+
+    database.close();
```

Comment:
Don't close the database connection manually, which might be shared within the same thread.